src/main/java/de/tum/in/www1/artemis/web/rest/dto/ExerciseDetailsDTO.java

```diff
@@ -0,0 +1,15 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.hestia.ExerciseHint;
+import de.tum.in.www1.artemis.service.iris.dto.IrisCombinedSettingsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.plagiarism.PlagiarismCaseInfoDTO;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record ExerciseDetailsDTO(Exercise exercise, IrisCombinedSettingsDTO irisSettings, PlagiarismCaseInfoDTO plagiarismCaseInfo, Set<ExerciseHint> availableExerciseHints,
+        Set<ExerciseHint> activatedExerciseHints) {
```

Comment:
Why do we not convert the ExerciseHints into DTOs as well?

Reply:
This PR doesn't contain any conversion. We only use a wrapper DTO. The conversion of the Exercise model to a DTO got scrapped for now.